### ISSUE TITLE
WS - Fatal error while updating a currency wia API

### ISSUE THREAD

**Issue by khouloudbelguith** [PrestaShop/PrestaShop#18865]
#### Describe the bug

When we try to update a currency via API, we have a fatal error
`Fatal error: Uncaught PrestaShopWebserviceException: This call to PrestaShop Web Services failed and returned an HTTP status of 400. That means: Bad Request. in /projet/ws/PSWebServiceLibrary.php:91 Stack trace: #0 /projet/ws/PSWebServiceLibrary.php(359): PrestaShopWebservice->checkStatusCode(400) #1 /projet/ws/Updatecurrency.php(28): PrestaShopWebservice->edit(Array) #2 {main} thrown in /projet/ws/PSWebServiceLibrary.php on line 91`

With PS1.7.6.2 => OK

#### Expected behavior

There should be no error.

#### Steps to Reproduce

On a clean shop, using [the WS library](https://github.com/PrestaShop/PrestaShop-webservice-lib), execute the following script. Don't forget to replace the URL and api key with your own.

```
<html><head><title>pUT DATA</title></head><body>
<?php
define('DEBUG', true);
ini_set('display_errors','on');
define('PS_SHOP_PATH', 'shop.com');
define('PS_WS_AUTH_KEY', 'Key');
require_once('PSWebServiceLibrary.php');
$ws = new PrestaShopWebservice(
    'http://shop.com',
    'Key',
    false);
$id = 2; //ID currency Here
$xml = $ws->get(array('resource' => 'currencies', 'id' => $id));
echo "BEFORE\n";
echo $xml->asXML();
unset($xml->currency->symbol);
$xml->currency->name= "EDIT Dollar";
$opt['putXml'] = $xml->asXML();
$opt['id'] = $id;
$opt['resource'] = 'currencies';
$xml = $ws->edit($opt);
$xml = $ws->get(array('resource' => 'currencies', 'id' => $id));
echo "AFTER\n";
echo $xml->asXML();
```

**Screenshots**
![image](https://user-images.githubusercontent.com/16067358/80480975-e2e16500-8951-11ea-8d93-3b8e7ecef2f9.png)

#### Additional information

* PrestaShop version: 1.7.6.5, 1.7.6.4, 1.7.7.x
* PHP version: 7.2

**Comment by khouloudbelguith** [comment]
Could be related to this PR: https://github.com/PrestaShop/PrestaShop/pull/17069

**Comment by Progi1984** [comment]
Fixed by #18969
